From e2261936c802ba98d48418078d58f9372b1fa401 Mon Sep 17 00:00:00 2001
From: Frank Wunderlich <frank-w@public-files.de>
Date: Sat, 20 Apr 2024 10:41:58 +0200
Subject: [PATCH 33/93] arm64: make rtc and eeprom work on r4

arm64: dts: disable rtc interrupt as it causes warning in gic driver

[    1.935187] WARNING: CPU: 0 PID: 1 at drivers/irqchip/irq-gic-v3.c:1706
gic_irq_domain_select+0x15c/0x204

fix rtc by using address 0x51
---
 .../boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
index 7cebb1b4f629..de138930bf10 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
@@ -250,6 +250,14 @@ i2c@0 { //eeprom,rtc,ngff
 			#size-cells = <0>;
 			reg = <0>;
 
+			pcf8563: rtc@51 {
+				compatible = "nxp,pcf8563";
+				reg = <0x51>;
+				//interrupts = <&pio 6 IRQ_TYPE_LEVEL_LOW>;
+				#clock-cells = <0>;
+				//status = "disabled";
+			};
+
 			eeprom@57 {
 				compatible = "atmel,24c02";
 				reg = <0x57>;
@@ -258,13 +266,6 @@ eeprom@57 {
 				size = <256>;
 			};
 
-			pcf8563: rtc@50 {
-				compatible = "nxp,pcf8563";
-				reg = <0x50>;
-				interrupts = <&pio 6 IRQ_TYPE_LEVEL_LOW>;
-				#clock-cells = <0>;
-				status = "disabled";
-			};
 		};
 
 		i2c_sfp1: i2c@1 {
-- 
2.49.0

